File: services/prompt_service/prompt_service.py
"""
提示词管理服务
提供提示词模板相关的业务逻辑处理
"""

from typing import List, Optional, Dict, Any
from .prompt_models import PromptTemplate, PromptCreate, PromptUpdate, PromptQuery
from .prompt_database import get_prompt_database

class PromptService:
    """提示词模板服务类"""
    
    def __init__(self):
        self.db = None

    async def _get_db(self):
        """获取数据库实例"""
        if self.db is None:
            self.db = await get_prompt_database()
        return self.db

    async def create_prompt(self, prompt_data: PromptCreate, created_by: str = None) -> PromptTemplate:
        """创建提示词模板"""
        db = await self._get_db()
        return await db.create_prompt(prompt_data, created_by)

    async def get_prompt_by_id(self, prompt_id: str) -> Optional[PromptTemplate]:
        """根据ID获取提示词模板"""
        db = await self._get_db()
        return await db.get_prompt_by_id(prompt_id)

    async def get_prompt_by_title(self, title: str) -> Optional[PromptTemplate]:
        """根据标题获取提示词模板"""
        db = await self._get_db()
        return await db.get_prompt_by_title(title)

    async def update_prompt(self, prompt_id: str, update_data: PromptUpdate, updated_by: str = None) -> Optional[PromptTemplate]:
        """更新提示词模板"""
        db = await self._get_db()
        return await db.update_prompt(prompt_id, update_data, updated_by)

    async def delete_prompt(self, prompt_id: str) -> bool:
        """删除提示词模板"""
        db = await self._get_db()
        return await db.delete_prompt(prompt_id)

    async def get_prompt_list(self, query: PromptQuery) -> Dict[str, Any]:
        """获取提示词模板列表"""
        db = await self._get_db()
        return await db.get_prompt_list(query)

    async def search_prompt(self, keyword: str) -> List[PromptTemplate]:
        """搜索提示词模板"""
        if not keyword.strip():
            raise ValueError("搜索关键词不能为空")
        
        db = await self._get_db()
        return await db.search_prompt(keyword)

    async def get_categories(self) -> List[str]:
        """获取所有分类"""
        db = await self._get_db()
        return await db.get_categories()

    async def get_prompt_types(self) -> List[str]:
        """获取提示词类型列表"""
        db = await self._get_db()
        return await db.get_prompt_types()

    async def get_popular_prompts(self, limit: int = 10) -> List[PromptTemplate]:
        """获取热门提示词模板"""
        db = await self._get_db()
        return await db.get_popular_prompts(limit)

    async def increment_view_count(self, prompt_id: str) -> bool:
        """增加查看次数"""
        db = await self._get_db()
        return await db.increment_view_count(prompt_id)

    async def increment_usage_count(self, prompt_id: str) -> bool:
        """增加使用次数"""
        db = await self._get_db()
        return await db.increment_usage_count(prompt_id)

    async def get_prompt_detail(self, prompt_id: str) -> Optional[PromptTemplate]:
        """获取提示词模板详情（自动增加查看次数）"""
        prompt = await self.get_prompt_by_id(prompt_id)
        if prompt:
            await self.increment_view_count(prompt_id)
        return prompt

    async def record_prompt_usage(self, prompt_id: str) -> bool:
        """记录提示词模板使用"""
        prompt = await self.get_prompt_by_id(prompt_id)
        if not prompt:
            return False
        
        return await self.increment_usage_count(prompt_id)

# 全局服务实例
_prompt_service: Optional[PromptService] = None

def get_prompt_service() -> PromptService:
    """获取提示词模板服务实例"""
    global _prompt_service
    if _prompt_service is None:
        _prompt_service = PromptService()
    return _prompt_service

File: frontend/src/services/promptService.ts
import { apiService } from './api';

export interface PromptTemplate {
  id?: string;
  title: string;
  content: string;
  description?: string;
  category?: string;
  tags: string[];
  type: 'system' | 'user' | 'assistant' | 'general';
  status: 'draft' | 'published' | 'archived';
  priority: number;
  view_count?: number;
  usage_count?: number;
  is_public: boolean;
  variables?: string;
  example_input?: string;
  example_output?: string;
  model_type?: string;
  temperature: number;
  max_tokens?: number;
  created_by?: string;
  updated_by?: string;
  created_at?: string;
  updated_at?: string;
}

export interface PromptQuery {
  page?: number;
  per_page?: number;
  keyword?: string;
  category?: string;
  status?: string;
  type?: string;
}

export interface PromptResponse {
  items: PromptTemplate[];
  total: number;
  page: number;
  per_page: number;
  pages: number;
}

export const promptService = {
  // 获取提示词模板列表
  async getPromptList(params: PromptQuery = {}) {
    return await apiService.get('/api/prompt/', params);
  },

  // 获取提示词模板详情
  async getPromptDetail(id: string) {
    return await apiService.get(`/api/prompt/${id}`);
  },

  // 创建提示词模板
  async createPrompt(data: Omit<PromptTemplate, 'id' | 'created_at' | 'updated_at' | 'view_count' | 'usage_count'>) {
    return await apiService.post('/api/prompt/', data);
  },

  // 更新提示词模板
  async updatePrompt(id: string, data: Partial<PromptTemplate>) {
    return await apiService.put(`/api/prompt/${id}`, data);
  },

  // 删除提示词模板
  async deletePrompt(id: string) {
    return await apiService.delete(`/api/prompt/${id}`);
  },

  // 搜索提示词模板
  async searchPrompt(keyword: string) {
    return await apiService.get('/api/prompt/search/', { keyword });
  },

  // 获取提示词模板分类
  async getPromptCategories() {
    return await apiService.get('/api/prompt/categories/list');
  },

  // 获取提示词模板类型
  async getPromptTypes() {
    return await apiService.get('/api/prompt/types/list');
  },

  // 获取热门提示词模板
  async getPopularPrompts(limit: number = 10) {
    return await apiService.get('/api/prompt/popular/list', { limit });
  },

  // 记录提示词模板使用
  async recordPromptUsage(id: string) {
    return await apiService.post(`/api/prompt/${id}/usage`);
  }
};
